File: packages/core/stages/07_tts/out.ts
import { SplitAudioTiming } from "../06_split_audio/out";

// start_ms: number; // split_audio start_ms（视频意图位置）
// end_ms: number; // start_ms + tts_duration_ms（TTS 实际结束位置）
export interface TtsSegment extends SplitAudioTiming {
  slot_end_ms: number; // split_audio end_ms（原始槽位终点，参考）
  tts_duration_ms: number;
  status: "success" | "skipped" | "error" | "empty";
}

export interface TtsFile {
  segments: TtsSegment[];
}

File: packages/core/src/stages/tts/out.rs
//! tts 阶段输出结构 (镜像 TS `packages/core/stages/07_tts/out.ts`)。

use serde::{Deserialize, Serialize};

use crate::stages::split_audio::out::SplitAudioTiming;

/// 单条 TTS 段 (镜像 TS `TtsSegment` = SplitAudioTiming + tts 字段)。
#[derive(Debug, Clone, Default, Serialize, Deserialize, specta::Type)]
pub struct TtsSegment {
    #[serde(flatten)]
    pub timing: SplitAudioTiming,
    /// split_audio end_ms (原始槽位终点, 参考)
    pub slot_end_ms: u64,
    /// TTS 生成音频时长
    pub tts_duration_ms: u64,
    /// 状态: success / skipped / error / empty
    pub status: String,
}

/// `tts/tts.json` (镜像 TS `TtsFile`)。
#[derive(Debug, Clone, Default, Serialize, Deserialize, specta::Type)]
pub struct TtsFile {
    pub segments: Vec<TtsSegment>,
}
